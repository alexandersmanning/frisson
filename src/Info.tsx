import React from "react";
import { syllable } from "syllable";

const countSyllables = (text: string) => {
  try {
    return syllable(text);
  } catch (error) {
    console.error("Error counting syllables:", error);
    return 0;
  }
};

export default function Info({ text }) {
  const word_count = text.trim().split(/\s+/).length;
  const syllable_count = countSyllables(text.trim());
  return (
    <div className="text-sm xl:text-md">
      <p>
        {word_count}
        {' '}
        <span className="text-gray-400">words</span>
      </p>
      <p>
        {syllable_count}
        {' '}
        <span className="text-gray-400">syllables</span>
      </p>
    </div>
  );
}
