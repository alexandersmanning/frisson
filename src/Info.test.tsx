import renderer from "react-test-renderer";
import Info from "./Info";
import React from "react";

it("shows the info panel", () => {
  const component = renderer.create(<Info text="hi there syllables" />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
