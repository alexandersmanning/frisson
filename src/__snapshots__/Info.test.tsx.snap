// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows the info panel 1`] = `
<div
  className="text-sm xl:text-md"
>
  <p>
    3
     
    <span
      className="text-gray-400"
    >
      words
    </span>
  </p>
  <p>
    5
     
    <span
      className="text-gray-400"
    >
      syllables
    </span>
  </p>
</div>
`;
